msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/mam/TestCase2506936.kt

```diff
@@ -0,0 +1,115 @@
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client.msal.automationapp.testpass.broker.mam
+
+import com.microsoft.identity.client.msal.automationapp.R
+import com.microsoft.identity.client.msal.automationapp.testpass.broker.AbstractMsalBrokerTest
+import com.microsoft.identity.client.ui.automation.annotations.SupportedBrokers
+import com.microsoft.identity.client.ui.automation.app.TeamsApp
+import com.microsoft.identity.client.ui.automation.broker.BrokerCompanyPortal
+import com.microsoft.identity.client.ui.automation.broker.IMdmAgent
+import com.microsoft.identity.client.ui.automation.installer.LocalApkInstaller
+import com.microsoft.identity.client.ui.automation.interaction.FirstPartyAppPromptHandlerParameters
+import com.microsoft.identity.client.ui.automation.interaction.PromptParameter
+import com.microsoft.identity.labapi.utilities.client.LabQuery
+import com.microsoft.identity.labapi.utilities.constants.ProtectionPolicy
+import com.microsoft.identity.labapi.utilities.constants.TempUserType
+import com.microsoft.identity.labapi.utilities.constants.UserType
+import org.junit.Test
+
+// TrueMAM: Sign In with Teams and then SignOut and Sign Back In
+// https://identitydivision.visualstudio.com/Engineering/_workitems/edit/2506936
+@SupportedBrokers(brokers = [BrokerCompanyPortal::class])
+class TestCase2506936 : AbstractMsalBrokerTest(){
+
+    @Test
+    fun test_2506936() {
+        // Fetch credentials
+        val username: String = mLabAccount.username
+        val password: String = mLabAccount.password
+
+        val teams = TeamsApp(LocalApkInstaller())
+        teams.install()
+        teams.launch()
+        teams.handleFirstRun()
+
+        val teamsPromptHandlerParameters = FirstPartyAppPromptHandlerParameters.builder()
+            .prompt(PromptParameter.SELECT_ACCOUNT)
+            .loginHint(username)
+            .broker(mBroker)
+            .registerPageExpected(true)
+            .enrollPageExpected(false)
+            .consentPageExpected(false)
+            .speedBumpExpected(false)
+            .build()
+
+        // Sign in the first time
+        teams.addFirstAccount(username, password, teamsPromptHandlerParameters)
+        // handle app protection policy in CP i.e. setup PIN when asked
+        (mBroker as IMdmAgent).handleAppProtectionPolicy()
+        teams.onAccountAdded()
+        teams.forceStop() // Teams sometimes seems to like to pop up on screen randomly
+
+        teams.signOut()
+
+        // Sign in again
+        val teamsPromptHandlerParameters2 = FirstPartyAppPromptHandlerParameters.builder()
+            .prompt(PromptParameter.SELECT_ACCOUNT)
+            .loginHint(username)
+            .broker(mBroker)
+            .registerPageExpected(false)
+            .enrollPageExpected(false)
+            .consentPageExpected(false)
+            .speedBumpExpected(false)
+            .secondPasswordPageExpected(true)
```

Comment:
Technically this is a second interactive request made by Teams (behind the scenes) and we could alternatively tackle this by creating a second prompt handler parameters and handle prompt (rather than having this secondPasswordPageExpected boolean), but I guess this is fine either way

Reply:
I see. We could go either way, but perhaps this approach makes things more streamlined?

Reply:
It seems it's unclear whether or not the password is prompted, i've had it only prompt once, seems inconsistent?